Source/WebKit/UIProcess/API/glib/WebKitDownload.cpp

```diff
@@ -449,52 +456,73 @@ WebKitURIRequest* webkit_download_get_request(WebKitDownload* download)
  * webkit_download_get_destination:
  * @download: a #WebKitDownload
  *
- * Obtains the URI to which the downloaded file will be written.
+ * Obtains the destination to which the downloaded file will be written.
  *
- * You
- * can connect to #WebKitDownload::created-destination to make
+ * You can connect to #WebKitDownload::created-destination to make
  * sure this method returns a valid destination.
  *
- * Returns: the destination URI or %NULL
+ * Returns: (nullable): the destination or %NULL
  */
 const gchar* webkit_download_get_destination(WebKitDownload* download)
 {
-    g_return_val_if_fail(WEBKIT_IS_DOWNLOAD(download), 0);
+    g_return_val_if_fail(WEBKIT_IS_DOWNLOAD(download), nullptr);
 
-    return download->priv->destinationURI.data();
+#if !ENABLE(2022_GLIB_API)
+    if (download->priv->destinationURI)
+        return download->priv->destinationURI.get();
+#endif
+    return download->priv->destination.get();
```

Comment:
OK, this seems almost good, but problem is applications that never call webkit_download_set_destination() might still be expecting a URI here. So to be backwards-compatible, I think we need to invert the logic in the `!ENABLE(2022_GLIB_API)` case: return a path only if a path is set with webkit_download_set_destination(), and return a URI otherwise. OK?

Reply:
We would still receive a URI in hat case too, because we set the destinationURI in webkitDownloadDecideDestination

Reply:
I've added a test case to cover this.

Reply:
OK, and I see you also added a test to make sure it works properly. Great.